backend/flaskr/model.py

```diff
@@ -15,35 +17,27 @@ def __init__(self, method, data):
         THIS CONSTRUCTOR IS "PRIVATE" AND SHOULD NOT BE CALLED.
         Initialization logic is handled in static factory methods.
         Description of fields:
-            num_requests = number of times this model has been used to generate a suggestion
-            food_genres  = dict of floats for how much a user likes each key [-10,10]
-            importances  = dict of floats for which restaurant features are most important {1-10}
+            num_reviews = number of reviews applied to this model
+            food_genres  = dict of dicts:
+                count      = number of interactions the user has had this category
+                propensity = floats for how much the user likes each key [-10,10]
+            importances  = dict of floats for which restaurant features are most important [0,10]
         """
         if method == "state":
-            self.num_requests = data["num_requests"]
+            self.num_reviews = data["num_reviews"]
             self.food_genres = data["food_genres"]
             self.importances = data["importances"]
 
         elif method == "quiz":
-            self.num_requests = 0
-            self.food_genres = {
-                data.pop("favorite"): 7,
-                data.pop("leastFavorite"): -7,
-            }
-            self.importances = {k: 2 * int(v) for k,v in data.items()}
+            self.num_reviews = 0
+            self.food_genres = create_genre_dict(data.pop("favorite"), data.pop("leastFavorite"))
+            self.importances = {k: max(0, min(10, 2 * int(v) - 1)) for k,v in data.items()}
```

Comment:
What does the -1 do here? If ratings go from 1-5, we can never hit the maximum of 10. Should it be min(9, 2 * int(v) - 1) instead?

Reply:
Technically yea. I just clamp is here because in the front end, you can type in numbers much higher and lower than 1-5 even though we *sort of* limit it. It definitely wouldn't hurt to clamp between 1-9 instead since those are the only possible values from the quiz.